### ISSUE TITLE
lowercase variable name causes false positive shellcheck warning

### ISSUE THREAD

**Issue by bigwheel** [rhysd/actionlint#53]
# minimum reproducing example

```example.yml
on: push
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - run: echo "$FOO"
        env:
          FOO: FOO
      - run: echo "$bar"
        env:
          bar: bar
```

```sh
❯ actionlint testdata/examples/shellcheck_integration.yaml
testdata/examples/shellcheck_integration.yaml:9:9: shellcheck reported issue in this script: SC2154:warning:1:7: bar is referenced but not assigned [shellcheck]
  |
9 |       - run: echo "$bar"
  |         ^~~~


❯ cat /etc/os-release
NAME="Ubuntu"
VERSION="21.04 (Hirsute Hippo)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 21.04"
VERSION_ID="21.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute


❯ actionlint --version
1.6.4
installed by downloading from release page
built with go1.17.1 compiler for linux/amd64
```

# What's happen

Actionlint doesn't give `env:`, `jobs.<job_id>.env` or `jobs.<job_id>.steps[*].env` to shellcheck.
[And shellcheck skips variable unassigned check only if it doesn't contains lowercase](https://github.com/koalaman/shellcheck/blob/master/CHANGELOG.md#v036---2015-03-28).

# Solution candidate

1. Generate variable assign code by `env:`, `jobs.<job_id>.env` and `jobs.<job_id>.steps[*].env` then prepend it to the `run` script.
2. [Ignore SC2154](https://github.com/rhysd/actionlint/blob/74ea0121a22d003bb3e8b6cab04c6cee6db70976/rule_shellcheck.go#L167-L167) (option) and write this limitation in README

**Comment by rhysd** [comment]
I'm sorry for late response. I was focusing on other project.

**Comment by bigwheel** [comment]
@rhysd Thanks responsible response!

**Comment by rhysd** [comment]
This fix was released at v1.6.5.
